.changeset/wild-melons-study.md

```diff
@@ -0,0 +1,93 @@
+---
+"cojson": minor
+"cojson-core-rn": minor
+"jazz-tools": minor
+---
+
+## 💥 Breaking Changes
+
+### Crypto provider cleanup
+
+- **Removed `PureJSCrypto`** from `cojson` (including the `cojson/crypto/PureJSCrypto` export)
+- **Removed `RNQuickCrypto`** from `jazz-tools`
+- **`RNCrypto` (Rust via `cojson-core-rn`) is now the default** on React Native + Expo
+- **`WasmCrypto` is the default** on Web / SSR / Workers and **no longer falls back** to JS crypto (WASM init failures now throw)
+
+Full migration guide: `https://jazz.tools/docs/upgrade/0-20-0`
+
+## Migration
+
+### React Native / Expo: `cojson-core-rn` is required
+
+Install and keep the version aligned with `jazz-tools`:
+
+```bash
+npm install jazz-tools cojson-core-rn
+```
+
+Docs:
+- React Native setup: `https://jazz.tools/docs/react-native/project-setup`
+- Expo setup: `https://jazz.tools/docs/react-native-expo/project-setup`
+- Providers: `https://jazz.tools/docs/project-setup/providers`
+
+### Recommended: remove `CryptoProvider` (no `RNCrypto` import needed)
+
+React Native:
+
+```tsx
+// BEFORE
+import { JazzReactNativeProvider } from "jazz-tools/react-native";
+import { RNCrypto } from "jazz-tools/react-native-core/crypto";
+
+<JazzReactNativeProvider sync={...} CryptoProvider={RNCrypto}>
+  {children}
+</JazzReactNativeProvider>
+
+// AFTER (recommended)
+import { JazzReactNativeProvider } from "jazz-tools/react-native";
+
+<JazzReactNativeProvider sync={...}>
+  {children}
+</JazzReactNativeProvider>
+```
+
+Expo:
+
+```tsx
+// BEFORE
+import { JazzExpoProvider } from "jazz-tools/expo";
+import { RNCrypto } from "jazz-tools/expo/crypto";
+
+<JazzExpoProvider sync={...} CryptoProvider={RNCrypto}>
+  {children}
+</JazzExpoProvider>
+
+// AFTER (recommended)
+import { JazzExpoProvider } from "jazz-tools/expo";
+
+<JazzExpoProvider sync={...}>
+  {children}
+</JazzExpoProvider>
+```
+
+### If you used `RNQuickCrypto`: switch to `RNCrypto` (or remove `CryptoProvider`)
+
+```ts
+// BEFORE
+import { RNQuickCrypto } from "jazz-tools/react-native-core/crypto";
+
+// AFTER
+import { RNCrypto } from "jazz-tools/react-native-core/crypto";
+```
+
+## SSR / Workers (WASM)
+
+WASM crypto no longer falls back to JS crypto. If you want async initialization:
```

Comment:
The change to a sync intialization is a minor, not even worth to be mentioned.

It's more imprtant that the lack of fallback to PureJSCrypto is going to break Cloudflare Workers.

It should be something like this:
> Before this release we were falling back to Javascript when the runtime doesn't support our default way of loading WASM (e.g. in Cloudflare Workers)
> This wasn't great because that fallback was coming with a 20x slowdown
> Now the load is going to fail with an error message that will guide you to the right solution
> More details in https://jazz.tools/docs/react/server-side/setup#wasm-on-edge-runtimes

Reply:
I also think that the instruction on how to run Jazz on edge runtimes should be more visible

cc  @joeinnes